### PR TITLE
Upgrade Dependencies (2020-11)

### PR DESCRIPTION
**Pull request by chenxsan**
Keeping dependencies up to date is beneficial to the health of this project, hence I suggest doing it every month.

- [x] replace [deprecated `request-promise`](https://github.com/request/request-promise#readme) with [alternative libraries](https://github.com/request/request/issues/3143)
- [x] upgrade react to v17 and enable [the new jsx transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) 
- [x] replace `node-sass` with `sass` as the former [was deprecated](https://github.com/sass/node-sass#node-sass) in favor of the latter
- [x] upgrade remark to 13 (breaking change)

### CODE REVIEW

**Comment by chenxsan** [comment]
https://github.com/webpack/webpack.js.org/pull/3963#issuecomment-706089344

> > So far the development server takes about 12s to start while recompiling takes about 3s. It would be great if we can improve the development speed which saves a lot of time for the contributors.

> Ok you have a pretty fast computer. For me it takes 56s initially and 9.6s for recompilation. And 19s for an initial build with persistent cache available.

Seems the initial performance has been improved compared to last month:

image

File: package.json [hunk 1]
```diff
@@ -172,6 +170,8 @@
     "webpack.vote": "https://github.com/webpack/voting-app.git"
   },
   "resolutions": {
-    "remark-responsive-tables": "git://github.com/chenxsan/remark-responsive-tables.git#bugfix/fix-empty-head"
+    "remark-responsive-tables": "git://github.com/chenxsan/remark-responsive-tables.git#bugfix/fix-empty-head",
+    "sitemap-static/minimist": "1.2.5",
```
Comment (montogeek) [inline package.json:174]: Why are the last 2 necessary?

File: src/utilities/content-tree-enhancers.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const remark = require('remark');
 const slug = require('remark-slug');
 const extractAnchors = require('remark-extract-anchors');
 const remarkHtml = require('remark-html');
+const frontmatter = require('remark-frontmatter');
```
Comment (montogeek) [inline src/utilities/content-tree-enhancers.js:8]: What changed that this is now necessary?

File: src/utilities/fetch-package-readmes.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const fs = require('fs');
 const path = require('path');
 const { promisify } = require('util');
 const mkdirp = require('mkdirp');
-const request = require('request-promise');
+const fetch = require('node-fetch');
```
Comment (montogeek) [inline src/utilities/fetch-package-readmes.js:6]: Why this change?

File: webpack.common.js [hunk 4]
```diff
@@ -131,13 +131,9 @@ module.exports = (env = {}) => ({
...
       {
         test: /\.woff2?$/,
-        use: {
-          // TODO use type: asset/resource when mini-css bug regarding asset modules is fixed
-          loader: 'file-loader',
-          options: {
-            outputPath: 'font',
-            esModule: false
-          }
+        type: 'asset/resource',
```
Comment (montogeek) [inline webpack.common.js:134]: Nice!

File: .babelrc [hunk 5]
```diff
@@ -1,7 +1,12 @@
 {
   "presets": [
     "@babel/preset-env",
-    "@babel/preset-react"
+    [
+      "@babel/preset-react",
+      {
+        "runtime": "automatic"
```
Comment (montogeek) [inline .babelrc:7]: Please remove all imports in the components.

File: src/utilities/fetch-package-readmes.js (hunk 3 above)
Comment (montogeek) [reply src/utilities/fetch-package-readmes.js:6]: Just read the PR description.

File: package.json (hunk 1 above)
Comment (chenxsan) [reply package.json:174]: These two lock their version of `minimist`, and there're vulnerabilities https://snyk.io/vuln/npm:minimist in lower version hence I upgrade them to the latest version.

File: src/utilities/content-tree-enhancers.js (hunk 2 above)
Comment (chenxsan) [reply src/utilities/content-tree-enhancers.js:8]: After upgrading remark to 13, there seems to be a bug in `remark-extract-anchors` when processing this article https://github.com/webpack/webpack.js.org/blob/master/src/content/blog/2020-10-10-webpack-5-release.md. It would parse the front matter as heading:

image

image

Which is wrong, hence I include `remark-frontmatter` to process the front matter first.

Comment (chenxsan) [reply src/utilities/content-tree-enhancers.js:8]: Ok, I found it can be easily fixed by deleting indentation from the front matter.

Before:

```yaml
---
title: Webpack 5 release (2020-10-10)
sort: -202010100
contributors:
    - sokra
    - chenxsan
---
```
After:

```yaml
---
title: Webpack 5 release (2020-10-10)
sort: -202010100
contributors:
  - sokra
  - chenxsan
---
```
Anyway, I believe including a remark plugin is much safer here as it can prevent potential issues in future.

**Comment by chenxsan** [comment]
@montogeek hi there, can you help take a look at the Vercel build log for me? I have no idea why it kept failing and I can't inspect this url  https://vercel.com/webpack-docs/webpack-js-org/1j77bd5rh.

**Comment by chenxsan** [comment]
I'm closing this in favor of other small pull requests as I don't have vercel access to inspect the reason of deployment failure.
